app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsFragment.kt

```diff
@@ -93,10 +94,16 @@ class SessionDetailsFragment : Fragment() {
         R.id.menu_item_add_to_calendar to { addToCalendar() },
         R.id.menu_item_flag_as_favorite to { favorSession() },
         R.id.menu_item_unflag_as_favorite to { unfavorSession() },
-        R.id.menu_item_set_alarm to { setAlarm() },
         R.id.menu_item_delete_alarm to { deleteAlarm() },
         R.id.menu_item_close_session_details to { closeDetails() },
         R.id.menu_item_navigate to { navigateToRoom() },
+        R.id.menu_item_set_alarm to {
+            if(NotificationHelper(requireContext()).notificationsEnabled) {
+                setAlarm()
+            } else {
+                Toast.makeText(requireContext(), resources.getString(R.string.alarm_disabled_toast), Toast.LENGTH_LONG).show()
+            }
+        },
```

Comment:
:building_construction: No shortcuts please. The code here contains business logic which should be located in the ViewModel. Please stick to the idea of unidirectional data flow, e.g. 

event in Fragment -> call ViewModel -> execute logic -> retrieve state (`notificationsEnabled`), split execution path (`if/else`) -> update view (Fragment) via observable property -> do what is view specific in the view (e.g. `Toast`)

I know it looks like an overhead to redirect the action through the ViewModel but this architecture allows you to test whatever happens in the view model and let the view be stupid. When the logic part becomes more complex this pays back.

Reply:
Yeah, I tried that first, but I think I didn't do that because I would have had to pass alot of stuff over to the View model. I'll move the code Over :+1:

Reply:
Try to follow the pattern used for other menu actions here and let me know when you got stuck.

Reply:
Shown in https://github.com/isiko/EventFahrplan/pull/1. Please have a look.